proposals/1984_ca-bundle-from-secret-ref/README.md

```diff
@@ -11,124 +11,261 @@ Once approved and implemented, they become historical documents.
-  * Code: `internal/controller/supervisorconfig/activedirectoryupstreamwatcher/active_directory_upstream_watcher.go`
-  * Status Condition: `TLSConfigurationValid`
-* `GitHubIdentityProvider.spec.githubAPI.tls.certificateAuthorityData`
-  * Code: `internal/controller/supervisorconfig/githubupstreamwatcher/github_upstream_watcher.go`
-  * Status Condition: `TLSConfigurationValid`
-* `LDAPIdentityProvider.spec.tls.certificateAuthorityData`
-  * Code: `internal/controller/supervisorconfig/ldapupstreamwatcher/ldap_upstream_watcher.go`
-  * Status Condition: `TLSConfigurationValid`
-* `OIDCIdentityProvider.spec.tls.certificateAuthorityData`
-  * Code: `internal/controller/supervisorconfig/oidcupstreamwatcher/oidc_upstream_watcher.go`
-  * Status Condition: currently `OIDCDiscoverySucceeded` appears to be the applicable status condition, but we should
-  add a new status condition `TLSConfigurationValid` to resemble the other identity provider CRs.
+Below is an example of trust-manager's bundle resource and the corresponding target of type `configMap`.
 
-### Concierge
+```bash
+apiVersion: trust.cert-manager.io/v1alpha1
+kind: Bundle
+metadata:
+  name: some-pinniped-test-cm
+spec:
+  sources:
+  - inLine: |
+      -----BEGIN CERTIFICATE-----
+      MIIOfDCC...
+      -----END CERTIFICATE-----
+  target:
+    configMap:
+      key: "foo.bar.baz"
+
+# results in a configMap with name some-pinniped-test-cm
+
+apiVersion: v1
+data:
+  foo.bar.whatever: |
+    -----BEGIN CERTIFICATE-----
+    MIIOfDCC...
+    -----END CERTIFICATE-----
+kind: ConfigMap
+metadata:
+  labels:
+    trust.cert-manager.io/bundle: some-pinniped-test-cm
+  name: some-pinniped-test-cm
+  namespace: concierge
+  ownerReferences:
+  - apiVersion: trust.cert-manager.io/v1alpha1
+    blockOwnerDeletion: true
+    controller: true
+    kind: Bundle
+    name: some-pinniped-test-cm
+    uid: ca2fe559-fb0b-4bb0-ae75-1b9b43df1d4c
+  resourceVersion: "82195"
+  uid: 08c96a5a-f59b-4e76-b0e2-7132abe16411
+```
+
+#### Vault
+
+Based on the usage reported in [issues/1886](https://github.com/vmware-tanzu/pinniped/issues/1886) the CA trust bundle is externally sourced into the kubernetes cluster as a secret.
+
+
+In conclusion, almost all of the tooling available to handle certificate operations either generate a kubernetes secret or a configmap with the CA certificates and/or CA bundles.
+
+
+## Proposal
+
+Based on the brief survey of the certificate management tooling in the kubernetes ecosystem, having Pinniped custom resources carry reference to 
+either kubernetes secrets or configmaps, with certificate authority data, will allow cluster-operators to plumb certificate management tooling into Pinniped custom resources.
+
+## Implementation
+The custom resources have an optional `certificateAuthorityData` field to hold the CA bundle to be used for client-side TLS verification as part of the `TLSSpec` that is currently defined as follows.
+
+```go
```

Comment:
Leading indentation on these structs appears inconsistent. It may be due to copy/paste of tabs vs. spaces.

Reply:
Actually, maybe we should show this as YAML rather than Go structs. If we show Go structs then we should probably figure out all the correct kubebuilder tags to show on the structs, but that seems like too much detail for the proposal, and could be figured out during implementation.

Reply:
Updated to add yaml instead of go structs.